content/en/04-tutorials/01-pixie-101/07-kafka-monitoring.md

```diff
@@ -0,0 +1,182 @@
+:::
+
+> The **Kafka Producer Pods** and **Kafka Consumer Pods** tables confirm that the `invoicing` client is consuming messages at a slower speed than they are being produced.
+
+> The `shipping` client has sent 292 `Fetch` requests while the `invoicing` client has sent much fewer (only 192 `Fetch` requests). This again indicates that something is wrong.
+
+> When debugging Kafka problems, you may want to check the health of the clients involved. Pixie makes it easy to switch between higher-level Kafka system metrics and the lower-level client infra metrics.
+
+5. To check the health of the `invoicing` client, click on the `px-kafka/invoicing-*` pod name in the **POD** column of the **Kafka Consumer Pods** table.
+
+> Clicking on any pod name will take you to the `px/pod` script for that pod. This script shows an overview of the pod's CPU usage, network traffic and throughput, disk, and memory usage.
+
+> In this case, CPU utilization is low, which means resource starvation is not causing the `invoicing` client to consume messages at a slower rate.
+
+## Kafka Events with Metadata
+
+Let's inspect the raw Kafka requests flowing through the cluster.
+
+1. Select `px/kafka_data` from the script drop-down menu.
+
+> Pixie is able to [automatically trace](https://docs.px.dev/about-pixie/pixie-ebpf/) all messages flowing through your cluster, identify the ones using the Kafka protocol and parse the message metadata. This script shows a sample of the most recent Kafka events in the cluster.
+
+> For each record you can see the source, destination, request command, request and response, and the latency. Note that Pixie only shows the size of the payload, not the content, because it's usually too large.
+
+::: div image-xl relative
+<svg title='' src='use-case-tutorials/kafka/kafka-data.png'/>
+:::
+
+2. Click the `TIME_` column title to sort by time.
+
+> We can see that there are multiple Kafka request commands being traced by Pixie, such as `Produce`, `Fetch`, `OffsetCommit`, and `Heartbeat`.
+
+3. Click on a table row with the `Produce` `REQ_CMD` to inspect the parsed request and response body in JSON format.
+
+> Pixie parses the request body and gives us information on the name of the topic, what partition it is producing to and the total size of the message set. Inspecting the response shows us if there's any error code coming back from the Kafka broker, base offset, log append time, etc.
+
+This script is useful for inspecting a specific request or filtering all requests by a certain command (you'll need to [edit the PxL script](https://docs.px.dev/using-pixie/using-live-ui/#write-your-own-pxl-scripts-edit-an-existing-script) to do that).
+
+## Producer Consumer Latency
+
+Producer consumer latency is the time between when a producer pushes a message to a topic to when it is fetched by a consumer.
+
+It's important to monitor this latency because many incidents begin with consumer lag.
```

Comment:
Maybe also add a a comment about how large consumer lags one one component can mean it is out of sync with the other components.